README.md

```diff
@@ -8,4 +8,245 @@ are in the process of being changed and will certainly break your code.
 [travis-image]: https://travis-ci.org/census-instrumentation/opencensus-go.svg?branch=master
 [travis-url]: https://travis-ci.org/census-instrumentation/opencensus-go
 [godoc-image]: https://godoc.org/github.com/census-instrumentation/opencensus-go?status.svg
-[godoc-url]: https://godoc.org/github.com/census-instrumentation/opencensus-go
\ No newline at end of file
+[godoc-url]: https://godoc.org/github.com/census-instrumentation/opencensus-go
+
+# Go stats core library
+
+## Keys and tagging
+
+### To create/retrieve a key
+A key is defined by its name. To use a key a user needs to know its name and type (currently only keys of type string are supported. Later support for keys of type int64 and bool will be supported). Calling CreateKeyString(...) multiple times with the same name returns the same key.
+To create/retrieve a key the user calls:
+
+Create/retrieve key:
+
+    if err, key1 := tags.CreateKeyString("keyNameID1"); err != nil {
+        // handle error
+    }
+    ...
+
+    if err, key2 := tags.CreateKeyString("keyNameID2"); err != nil {
+        // handle error
+    }
+    ...
+
+### Create a set of tags associated with keys
+To create a new tag set from scratch using changes:
+
+    tsb := &tags.TagSetBuilder{}
+    tsb..StartFromEmpty()
+    tsb.InsertString( key1, "foo value")
+    tsb.UpdateString( key1, "foo value2")
+    tsb.UpsertString( key2, "bar value")
+    tagsSet := tsb.Build()
+
+    // A shorter way of achieving the same is:
+    tagsSet := &tags.TagSetBuilder{}.StartFromEmpty()
+                                    .InsertString( key1, "foo value")
```

Comment:
This snippet won't compile. In Go, try doing:

```
tagsSet := &tags.TagSetBuilder{}.StartFromEmpty().
                                     InsertString(key1, "foo value").
                                     UpdateString( key1, "foo value2").
etc
```

Reply:
Done